### PR TITLE
[RLlib] Revert PPO back to old API stack (by default). New stack and PPO not ready yet on several features.

### PR DESCRIPTION
**Pull request by sven1977**
1) Revert PPO back to old API stack (by default).
* PPO on the new stack is NOT ready yet on several features, including LSTM, disabling exploration (e.g. on the eval workers), attention net, trajectory view API.
* We will re-activate PPO on the new stack by default, once it has been fully moved to the EnvRunner APIs and supports multi-agent, connectors, and all the above mentioned currently missing functionalities.

2) Renamed config args: `_enable_rl_module_api` and `_enable_learner_api` into a single `_enable_new_api_stack` setting to remove confusion. These two settings already had to be either both switch on or both switched off anyways.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/algorithm_config.py [hunk 1]
```diff
@@ -896,22 +895,37 @@ def validate(self) -> None:
...
-                "False)."
+        # Throw a warning if the user has used rl_module_spec but not enabled the
+        # new API stack.
+        if self.rl_module_spec is not None and not self._enable_new_api_stack:
+            logger.warning(
+                "You have setup a RLModuleSpec (via calling `config.rl_module(...)`), "
+                "but have not enabled the new API stack. To enable it, call "
+                "`config.experimental(_enable_new_api_stack=True)`."
+            )
+        # LR-schedule checking.
```
Comment (sven1977) [inline rllib/algorithms/algorithm_config.py:913]: Moved here b/c it used to be in the wrong place.

File: rllib/algorithms/bc/bc.py [hunk 2]
```diff
@@ -116,11 +115,6 @@ def get_default_learner_class(self) -> Union[Type[Learner], str]:
 
     @override(MARWILConfig)
     def validate(self) -> None:
-        # Can not use Tf with learner api.
```
Comment (sven1977) [inline rllib/algorithms/bc/bc.py:119]: This is already checked in `validate()`. We should never(!) automatically change properties inside AlgorithmConfig (unless private ones that are covered by (public) @properties).

File: rllib/algorithms/algorithm_config.py [hunk 3]
```diff
@@ -462,13 +461,13 @@ def __init__(self, algo_class=None):
         self.worker_restore_timeout_s = 1800
 
         # `self.rl_module()`
-        self.rl_module_spec = None
-        self._enable_rl_module_api = False
+        self._rl_module_spec = None
```
Comment (sven1977) [inline rllib/algorithms/algorithm_config.py:464]: Made this private (plus a @property for `rl_module_spec`). We should never automatically (or inside `validate()`) set any properties. This now makes a lot of tests better as they don't require to magically wait for things to change under the hood after calling `validate()`.

File: rllib/algorithms/algorithm_config.py [hunk 4]
```diff
@@ -1784,9 +1768,6 @@ def training(
                 dashboard. If you're seeing that the object store is filling up,
                 turn down the number of remote requests in flight, or enable compression
                 in your experiment of timesteps.
-            _enable_learner_api: Whether to enable the LearnerGroup and Learner
```
Comment (kouroshHakha) [inline rllib/algorithms/algorithm_config.py:1787]: ~nit: don't remove the doc for it. Just add that it has been replaced with `_enable_new_stack_api`~

I see that you throw the deprecation warning right away.

**Review by kouroshHakha — APPROVED** [review]
LGTM

**Pull request merged**